### PR TITLE
Add macOS CI on GitHub Actions

### PR DESCRIPTION
**Pull request by williamfgc**
## Proposed changes

Try out macOS CI using GitHub Actions hosted runner
gcc-11 real build
homebrew dependencies
disregard some automated YAML formatting 

To do:
Add documentation if macOS CI passes

## What type(s) of changes does this code introduce?

- Testing changes (e.g. new unit/integration/performance tests) CI

### Does this introduce a breaking change?

- Yes

## What systems has this change been tested on?
Must pass tests on GitHub Actions macOS VM

## Checklist

- Yes. This PR is up to date with current the current state of 'develop'
- Yes. Code added or changed in the PR has been clang-formatted
- No. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- No. Documentation has been added (if appropriate)

### CODE REVIEW

**Comment by prckent** [comment]
This is still marked as WIP, but it looks good to me. I notice that the accelerate framework was correctly picked up.

**Comment by williamfgc** [comment]
@prckent  great, thanks! It's still WIP since I'd like to try the Python tests and add documentation, now that we have a better understanding of this CI workflow.

**Review by prckent — APPROVED** [review]
LGTM, will need a non ORNL person to merge.

**Review by ye-luo — DISMISSED** [review]

**Comment by williamfgc** [comment]
I'll just rebase `develop` one more time before testing.

**Review by ye-luo — APPROVED** [review]

**Comment by ye-luo** [comment]
Test this please

File: .github/workflows/ci-github-actions.yaml [hunk 1]
```diff
@@ -20,79 +18,112 @@ jobs:
...
-        flags: tests-deterministic # optional
-        name: codecov-QMCPACK # optional
-        fail_ci_if_error: true # optional (default = false)
+      - name: Checkout Action
+        uses: actions/checkout@v2
+
+      - name: Setup Dependencies
+        run: |
+             brew install ninja hdf5 fftw boost
+             pip3 install numpy h5py pandas
```
Comment (quantumsteve) [inline .github/workflows/ci-github-actions.yaml:117]: @williamfgc Homebrew has it's own numpy formula https://formulae.brew.sh/formula/numpy

Comment (williamfgc) [reply .github/workflows/ci-github-actions.yaml:117]: @quantumsteve thanks. @prckent @ye-luo @jtkrogel  would a typical QMCPACK macOS user have `pip3` or `brew` packages for nexus usage (that's the motivation for adding Python).

**Pull request merged**